src/providers/Gardens.js

```diff
@@ -0,0 +1,104 @@
+import React, { useContext, useEffect, useMemo, useState } from 'react'
+import { useRouteMatch } from 'react-router-dom'
+import { addressesEqual } from '@1hive/1hive-ui'
+import { getGardens } from '@1hive/connect-gardens'
+import daoList from '@1hive/gardens-dao-list'
+import { DAONotFound } from '../errors'
+
+import { useTokenBalances } from '../hooks/useOrgHooks'
+import { useWallet } from './Wallet'
+import { getNetwork } from '../networks'
+
+const DAOContext = React.createContext()
+
+export function GardensProvider({ children }) {
+  const [gardens, loading] = useGardensList()
+
+  const match = useRouteMatch('/garden/:daoId')
+
+  const connectedGarden = useMemo(() => {
+    if (match) {
+      const gardenAddress = match.params.daoId
+      return gardens.find(d => addressesEqual(gardenAddress, d.address))
+    }
+
+    return null
+  }, [gardens, match])
+
+  if (connectedGarden) {
+    return (
+      <WithConnectedGarden connectedGarden={connectedGarden} gardens={gardens}>
```

Comment:
wondering why in this if instead of returning this new `WithConnectedGarden` we just assign the connectedGarden variable? also doing with that way we avoid possible issues trying to deconstruct when using the useGardens hook and trying to access the .connectedGarden in some place that is not present, i think we should always return it and if does not exist return null

Reply:
The rationale for doing it this way was so we don't call the token balances hook when no garden is connected. If `connectedGarden` isn't present in some component that's using the `useGardens` hook, there's no real harm since from the structure that we've created, each component knows when a `connectedGarden` could or could not be present. 

That being said, in the next PR we'll lift up the Connect provider and AppState provider to the Gardens provider and encapsulate children with said providers only if there's a connected garden so components outside the `GardenHome` tree can access the garden's data. So the `WithConnectedGarden` won't be necessary